### DISCUSSION TITLE
Development and  Production Environment Best Practices

### DISCUSSION THREAD

**Discussion by fazal-kh4n** [sqlalchemy/alembic#1038] [Usage Questions]
Hi,

I currently have the following setup:

- A development environment where I add new features (i.e. upgrade the database schema).
- A production environment which should reflect changes in the development environment after they have been improved.

A typical workflow may be:

- Development Environment: Add a new column
- Test if the new column works etc
- Push this change downstream to the Production Environment, so that the new column is added there too.

What would be the best way to handle this workflow using Alembic?

Should I create separate branches for Development and Production, and simply merge the branches when the changes are ready to be deployed?

**Comment by CaselIT** [comment]
Hi

I believe what you have mentioned as workflow is pretty common.
Once the change are in the production environment you update the production db doing alembig upgrade.
Usually the db connection strings are passed as env variables or other similar things (like using a cloud secret management framework) so depending on that connection string the changes are done do the appropriate database.

> Should I create separate branches for Development and Production, and simply merge the branches when the changes are ready to be deployed?

I don't think it's necessary for the workflow you have described, since the database are different anyway

**Reply by fazal-kh4n** [reply]
Thanks a lot @CaselIT! That makes sense. I will take a deeper look in to branches etc. Those features look super helpful but I haven't quiete figured out where to use them.
